Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8392,6 +8393,28 @@ void WebPageProxy::decidePolicyForResponseShared(Ref<WebProcessProxy>&& process,
         m_navigationClient->decidePolicyForNavigationResponse(*this, WTFMove(navigationResponse), WTFMove(listener));
 }
 
+void WebPageProxy::showBrowsingWarning(RefPtr<WebKit::BrowsingWarning>&& safeBrowsingWarning)
+{
+    Ref protectedPageLoadState = pageLoadState();
+    auto transaction = protectedPageLoadState->transaction();
+    protectedPageLoadState->setTitleFromBrowsingWarning(transaction, safeBrowsingWarning->title());
+    protectedPageClient()->showBrowsingWarning(*safeBrowsingWarning, [protectedThis = Ref { *this }] (auto&& result) mutable {
+        Ref protectedPageLoadState = protectedThis->pageLoadState();
+        auto transaction = protectedPageLoadState->transaction();
+        protectedPageLoadState->setTitleFromBrowsingWarning(transaction, { });
+
+        switchOn(result, [protectedThis] (const URL& url) {
+            protectedThis->loadRequest(URL { url });
```

Comment:
Do we need to re-load the page here?

Reply:
This is used for loading the "report this page isn't actually bad" url.